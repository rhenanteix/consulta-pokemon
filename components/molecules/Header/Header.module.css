.header {
  align-items: center;
  background-color: var(--color-white);
  display: flex;
  height: 104px;
  justify-content: space-between;
  padding: 22px 83px;
  width: 100%;
}

.nav {
  align-items: center;
  display: flex;
  gap: 30px;
  list-style: none;
}

.link {
  color: var(--color-black);
  font-size: 14px;
  font-weight: 400;
  line-height: 17px;
}

@media screen and (max-width: 768px) {
  .header {
    padding: 16px;
    flex-wrap: wrap;
    height: auto;
    gap: 30px;
  }
}