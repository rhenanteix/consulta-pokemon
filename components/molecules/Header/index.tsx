import Link from 'next/link'
import { useRouter } from 'next/router'

import MenuPokeBall from '@/components/atoms/MenuPokeBall'
import Button from '@/components/atoms/Button'

import styles from './Header.module.css'

export default function Header() {
  const router = useRouter()

  return (
    <header className={styles.header}>

      <div className="pokeball-brand">
        <MenuPokeBall />
      </div>

      <ul className={styles.nav}>
        <li className={styles.item}>
          <Link className={styles.link} href='/quem-somos'>Quem Somos</Link>
        </li>
        <li className={styles.item}>
          <Button textLabel='Agendar Consulta' onClickEvent={() => router.push('/agendar-consulta')} />
        </li>
      </ul>
    </header>
  )
}