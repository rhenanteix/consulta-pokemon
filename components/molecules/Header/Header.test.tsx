import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'

import Header from '.'

jest.mock('next/router', () => require('next-router-mock'))

it('<Header /> tests', () => {
  const {
    container
  } = render(<Header />)

  expect(container).toMatchSnapshot()
})