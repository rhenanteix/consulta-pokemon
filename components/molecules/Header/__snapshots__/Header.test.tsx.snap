// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> tests 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="pokeball-brand"
    >
      <div
        class="menu open"
      >
        <img
          alt="Centro Pokémon"
          data-nimg="1"
          decoding="async"
          height="34"
          loading="lazy"
          src="/images/white-pokeball.svg"
          srcset="/images/white-pokeball.svg 1x, /images/white-pokeball.svg 2x"
          style="color: transparent;"
          width="34"
        />
        <p
          class="text"
        >
          Centro Pokémon
        </p>
      </div>
    </div>
    <ul
      class="nav"
    >
      <li
        class="item"
      >
        <a
          class="link"
          href="/quem-somos"
        >
          Quem Somos
        </a>
      </li>
      <li
        class="item"
      >
        <button
          class="btn"
        >
          Agendar Consulta
        </button>
      </li>
    </ul>
  </header>
</div>
`;
